py-polars/polars/series/series.py

```diff
@@ -268,17 +268,18 @@ def __init__(
                 version="0.20.6",
             )
 
-        # If 'Unknown' treat as None to attempt inference
+        # If 'Unknown' treat as None to trigger type inference
         if dtype == Unknown:
             dtype = None
-        # Raise early error on invalid dtype
-        elif (
-            dtype is not None
-            and not is_polars_dtype(dtype)
-            and py_type_to_dtype(dtype, raise_unmatched=False) is None
-        ):
-            msg = f"given dtype: {dtype!r} is not a valid Polars data type and cannot be converted into one"
-            raise ValueError(msg)
+        elif dtype is not None and not is_polars_dtype(dtype):
+            # Raise early error on invalid dtype
+            if not is_polars_dtype(
+                py_dtype := py_type_to_dtype(dtype, raise_unmatched=False)
```

Comment:
nitpicky: the variable `py_dtype` is a polars data type, so can we name this `pl_dtype` instead?

Reply:
Good spot; in my head I was thinking `from_py_dtype` and omitted the leading `from_` 😅